backend/model_server/main.py

```diff
@@ -32,22 +31,31 @@
 logger = setup_logger()
 
 
-async def manage_huggingface_cache() -> None:
-    temp_hf_cache = Path("/root/.cache/temp_huggingface")
-    hf_cache = Path("/root/.cache/huggingface")
-    if temp_hf_cache.is_dir() and any(temp_hf_cache.iterdir()):
-        hf_cache.mkdir(parents=True, exist_ok=True)
-        for item in temp_hf_cache.iterdir():
+def manage_huggingface_cache() -> None:
+    logger.info("Moving contents of temp_huggingface to huggingface cache.")
+    temp_hf_cache = Path("/root/.cache/temp_huggingface/hub/")
+    hf_cache = Path("/root/.cache/huggingface/hub/")
+
+    if not temp_hf_cache.is_dir():
+        return
+
+    # we have to move each file individually because the directories might
+    # have the same name but not the same contents
+    def _move_files_recursively(source: Path, dest: Path) -> None:
+        for item in source.iterdir():
             if item.is_dir():
-                await asyncio.to_thread(
-                    shutil.copytree, item, hf_cache / item.name, dirs_exist_ok=True
-                )
+                _move_files_recursively(item, dest / item.relative_to(source))
             else:
-                await asyncio.to_thread(shutil.copy2, item, hf_cache)
-        await asyncio.to_thread(shutil.rmtree, temp_hf_cache)
-        logger.info("Copied contents of temp_huggingface and deleted the directory.")
-    else:
-        logger.info("Source directory is empty or does not exist. Skipping copy.")
+                target = dest / item.relative_to(source)
```

Comment:
it seems like dest / item.relative_to(source) is used in both code paths. Can't this be lifted out?